### PR TITLE
fix: show output does not work, panel serializer does not work

### PR DESCRIPTION
**Pull request by slavik-lvovsky**

### CODE REVIEW

File: packages/backend/src/panels/CodeSnippetPanel.ts [hunk 1]
```diff
@@ -82,16 +92,20 @@ export class CodeSnippetPanel extends AbstractWebviewPanel {
...
 
-  private async prepareSnippet(uiOptions?: unknown): Promise<any> {
-    const snippet = await this.contributors.getSnippet(uiOptions);
+  private validateSnippet(snippet: unknown, contributorInfo: unknown) {
     if (_.isNil(snippet)) {
       const errorMessage = `'${_.get(
-        uiOptions,
+        contributorInfo,
         "contributorId"
       )}' snippet could not be found.`;
```
Comment (alex-gilin) [inline packages/backend/src/panels/CodeSnippetPanel.ts:100]: duplicate with getContribInfoFrom()

File: packages/backend/src/panels/CodeSnippetPanel.ts [hunk 2]
```diff
@@ -18,15 +18,27 @@ import { SnippetFlowPromise, RejectType, ResolveType } from "src/utils";
 export class CodeSnippetPanel extends AbstractWebviewPanel {
   public static CODE_SNIPPET = "Code Snippet";
   private static channel: vscode.OutputChannel;
-  private readonly panelPromise: Promise<void>;
+  private codeSnippetFlowPromise: Promise<void>;
   private snippetFlowPromise: SnippetFlowPromise;
 
+  // uiOptions comming from webview panel serializer looks like: {contributorInfo: {...}, ...}
+  private getContribInfoFrom(uiOptions?: unknown) {
```
Comment (alex-gilin) [inline packages/backend/src/panels/CodeSnippetPanel.ts:25]: signature - return value missing

File: packages/backend/src/panels/CodeSnippetPanel.ts [hunk 3]
```diff
@@ -82,16 +92,20 @@ export class CodeSnippetPanel extends AbstractWebviewPanel {
...
+  private validateSnippet(snippet: unknown, contributorInfo: unknown) {
     if (_.isNil(snippet)) {
       const errorMessage = `'${_.get(
-        uiOptions,
+        contributorInfo,
         "contributorId"
       )}' snippet could not be found.`;
       this.logger.error(errorMessage);
-      return Promise.reject(errorMessage);
+      throw new Error(errorMessage);
```
Comment (alex-gilin) [inline packages/backend/src/panels/CodeSnippetPanel.ts:102]: is the same like reject , isn't it ?

**Review by alex-gilin — CHANGES_REQUESTED** [review]

File: packages/backend/src/panels/CodeSnippetPanel.ts (hunk 1 above)
Comment (slavik-lvovsky) [reply packages/backend/src/panels/CodeSnippetPanel.ts:100]: No, I do not see any duplication here

File: packages/backend/src/panels/CodeSnippetPanel.ts (hunk 3 above)
Comment (slavik-lvovsky) [reply packages/backend/src/panels/CodeSnippetPanel.ts:102]: the effect is the same

File: packages/backend/src/panels/CodeSnippetPanel.ts (hunk 2 above)
Comment (slavik-lvovsky) [reply packages/backend/src/panels/CodeSnippetPanel.ts]: Done

**Review by alex-gilin — APPROVED** [review]

File: packages/backend/package.json [hunk 4]
```diff
@@ -31,9 +31,7 @@
     "sap"
   ],
   "activationEvents": [
-    "*",
-    "onCommand:loadCodeSnippet",
-    "onWebviewPanel:codeSnippet"
+    "*"
```
Comment (bd82) [inline packages/backend/package.json:34]: 👍

File: packages/backend/src/panels/CodeSnippetPanel.ts [hunk 5]
```diff
@@ -13,22 +13,42 @@ import { AbstractWebviewPanel } from "./AbstractWebviewPanel";
...
+import { createFlowPromise, FlowPromise } from "../utils";
 
 export class CodeSnippetPanel extends AbstractWebviewPanel {
   public static CODE_SNIPPET = "Code Snippet";
   private static channel: vscode.OutputChannel;
-  private readonly panelPromise: Promise<void>;
-  private snippetFlowPromise: SnippetFlowPromise;
+  private flowPromise: FlowPromise<void>;
+
+  // uiOptions comming from webview panel serializer looks like: {contributorInfo: {...}, ...}
```
Comment (bd82) [inline packages/backend/src/panels/CodeSnippetPanel.ts:23]: `coming` (single `m`)

File: packages/backend/src/panels/CodeSnippetPanel.ts [hunk 6]
```diff
@@ -13,22 +13,42 @@ import { AbstractWebviewPanel } from "./AbstractWebviewPanel";
...
   }
 
+  private setCommandPromiseAndFlowState() {
+    this.flowPromise = createFlowPromise();
+  }
+
+  private cleanCommandPromiseAndFlowState() {
+    if (this.flowPromise) {
+      // resolves command promise in case panel is closed manually by an user
+      // it is save to call resolve several times on same promise
```
Comment (bd82) [inline packages/backend/src/panels/CodeSnippetPanel.ts:39]: `save` --> `safe`

File: packages/backend/src/utils.ts [hunk 7]
```diff
@@ -1,7 +1,22 @@
-export type ResolveType = (value: void | PromiseLike<void>) => void;
-export type RejectType = (reason?: any) => void;
+type ResolveFunc<T> = (value: T | PromiseLike<T>) => void;
```
Comment (bd82) [inline packages/backend/src/utils.ts:1]: did not we locate some of these signatures is es5.d.ts (built-in types)?

Comment (bd82) [reply packages/backend/src/utils.ts:1]: never mind, it does not seem like the RejectType / ResolveType are easily extractable from there (or at all?)

File: packages/backend/src/panels/CodeSnippetPanel.ts (hunk 5 above)
Comment (slavik-lvovsky) [reply packages/backend/src/panels/CodeSnippetPanel.ts:23]: Done

File: packages/backend/src/panels/CodeSnippetPanel.ts (hunk 6 above)
Comment (slavik-lvovsky) [reply packages/backend/src/panels/CodeSnippetPanel.ts:39]: Done

**Pull request merged**